src/Compilers/Core/Portable/InternalUtilities/WeakList.cs

```diff
@@ -158,47 +158,68 @@ public void Add(T item)
             _items[_size++] = new WeakReference<T>(item);
         }
 
-        public IEnumerator<T> GetEnumerator()
+        public struct Enumerator
         {
-            int count = _size;
-            int alive = _size;
-            int firstDead = -1;
+            private readonly WeakList<T> _weakList;
+            private int _nextIndex;
+            private readonly int _count;
```

Comment:
move readonlys together.

Reply:
is this needed, or can we just defer to _weakList.Count (or is the concern that that might mutate?)

Reply:
ignore second comment.